src/protocols/hii.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright © 2019 Intel Corporation
+// Copyright (c) Microsoft Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(unused)]
+
+pub type StringId = u16;
+pub type HiiFontStyle = u32;
```

Comment:
Mh, there is not really an `HII` protocol, is there? Given that the HII packages are all defined in a global scope in the specification, I think we could justify putting this into `src/hii.rs`. So basically copying section `33.3` into `src/hii.rs` and dropping the `EFI_HII_` prefix.

I don't really mind copying only parts of the specification, but for consistency I try to copy entire sections whenever I need something from a specific section. This makes it easier to see which parts have been extracted. If this is a bit excessive for the HII sections, we can deviate. I have not spent much time with the HII interfaces, though.

Reply:
Running into a problem right away. HII has a bitfield in a structure and I'm not versed enough to confidently write the Rust bindings and don't want to introduce crate dependencies.

Thoughts on how to do this:
```c
// EFI_HII_PACKAGE_HEADER
// Summary - The header found at the start of each package.
// Prototype
    typedef struct {
        UINT32 Length:24;
        UINT32 Type:8;
        UINT8 Data[ … ];
    } EFI_HII_PACKAGE_HEADER;
```

Reply:
Does the UEFI specification declare its bitfield encoding somewhere? It is implementation defined in C, and it actually differs between compilers. Furthermore, endianness becomes an issue, if UEFI ever starts supporting big-endian machines.

However, I think the intent of this structure really is to have a 3-byte length field followed by an 1-byte type field. Given that those structures can actually be written to disk, I would even argue we can hard-code the little-endian behavior. But please correct me if I am wrong.

In that case, I think the easiest solution would be:

```rust
    pub length: [u32; 3],
    pub r#type: u8, // the `r#` syntax allows using reserved identifiers
```

And you would end up transmuting your `u32` to `[u32; 4]`, then subslice `[0..2]` and write it to `length`. The inverse operation would simply extend the `length` field to an `[u32; 4]` and transmute back to `u32`.

I currently do not care too much about endianness in `r-efi`, since the spec does not mention this anyway. And if big-endian ever becomes a thing in UEFI, you probably have to audit all the structures, anyway.